README.md

```diff
@@ -1,2 +1,213 @@
+  - Enforce naming conventions for variables and function names
+  - Use semicolons at the end of statements
+  - Use double quotes for strings
+  - Enforce no trailing spaces, consistent line endings (e.g., LF or CRLF), and consistent spacing around operators
+  - Limit the line length to a maximum of 100 characters
+  - Limit the max 300 lines for files
+  - Limit the max 30 lines for functions
+  - Enforce 2 space identation
+  - Setup prettier and format code with prettier
+  - Ignore checks in non-php files
+
+- Write GitLab CI script to run lint
+  - Create a repository on GitLab if one doesn't already exist
+  - Configure the Pre-commit hook to run prettier before the commit
+  - Configure the script to run Lint as part of the CI/CD pipeline.
+  - Ensure that the linting process runs on each commit.
+
+### Practical 3 - Linting Golang code
+
+**Expected -** The GitHub workflow job for Linting and Formatting should be passed without errors.
+
+- Set up Golang API project and learn Golang linters.
+  
+- Perform following rules,
+  - Enforce naming conventions for files and folder names
+  - Enforce naming conventions for variables and function names
+  - Limit the line length to a maximum of 100 characters
+  - Limit the max 300 lines for files
+  - Limit the max 30 lines for functions
+  - Format code
+  - Ignore check in non-go files
+  
+- Write GitHub CI script to run lint
+  - Create a repository on GitHub if one doesn't already exist
+  - Configure the Pre-commit hook to run prettier before the commit
+  - Configure the script to run Lint as part of the CI/CD pipeline.
+  - Ensure that the linting process runs on each commit.
+
+# Automate Testing
+
+### Practical 4 - Unit test check in golang
+
+**Expected -** The GitHub workflow job for unit test should be passed without errors.
+
+- Write unit tests from [Practical 22](https://github.com/canopas/web-developer-roadmap-2023#practical-22) if not already written.
+
+- Ensure the test covers all the success and error cases and passes with 85% code coverage.
+
+- Write GitHub CI script to run Unit test
+  - Create a repository on GitHub if one doesn't already exist
+  - Configure the database and script to run unit tests as part of the CI/CD pipeline.
+  - Ensure that the unit test runs on each commit.
+
+# Deployment
+
+## Running project using Docker on AWS EC2
+
+### Practical 5 - Build a Docker image
+
+**Expected -** Docker image should be available on GitLab Registry.
+
+- Set up [VueJs- admin panel](https://github.com/canopas/web-developer-roadmap-2023#practical-18) project if one doesn't already exist
+
+- Write GitLab CI script to run docker build
+  - Create a repository on GitLab if one doesn't already exist
+  - Write a docker file which includes,
+    - Install dependencies
+    - Build and run project
+  - Write a CI/CD script for Building a docker image and push it on the GitLab Registry.
+
+### Practical 6 - AWS EC2 Deployment
+
```

Comment:
Need to mention how to get AWS credentials or where to find it if don't want to create account.

Reply:
As this is an open-source project, I think it's not worth mentioning on AWS credentials. We can add `Must have AWS account` point in the prerequisites